### PR TITLE
[Glimmer2] Implement glimmer local lookup for components

### PR DESCRIPTION
**Pull request by asakusuma**
Replaces https://github.com/emberjs/ember.js/pull/13378

### CODE REVIEW

File: packages/ember-glimmer/lib/environment.js [hunk 1]
```diff
@@ -224,12 +227,15 @@ export default class Environment extends GlimmerEnvironment {
     return false;
   }
 
-  getComponentDefinition(path) {
+  getComponentDefinition(path, parentMeta = {}) {
```
Comment (krisselden) [inline packages/ember-glimmer/lib/environment.js]: Lets not default it here, lets either ensure it is set once on the template and always there or check presence

Comment (asakusuma) [inline packages/ember-glimmer/lib/environment.js]: @krisselden what would be the right way to get the parent meta info in the dynamic case?

So [here](https://github.com/emberjs/ember.js/blob/c8824b01ccc1105fc1e9191b23ac905efd760cd4/packages/ember-glimmer/lib/syntax/dynamic-component.js#L49) and [here](https://github.com/emberjs/ember.js/blob/c8824b01ccc1105fc1e9191b23ac905efd760cd4/packages/ember-glimmer/lib/syntax/dynamic-component.js#L12).

**Comment by asakusuma** [comment]
@krisselden I added [some tests](https://github.com/emberjs/ember.js/pull/13778/files#diff-60d0781473fafba06d2d55675ee1286cR81), which are failing, that illustrate the dynamic component local lookup case that I need guidance on.

File: packages/ember-glimmer/lib/environment.js [hunk 2]
```diff
@@ -198,18 +198,21 @@ export default class Environment extends GlimmerEnvironment {
       let internalKey = builtInComponents[key];
       let definition = null;
 
+      let template = templates && templates['default'];
```
Comment (krisselden) [inline packages/ember-glimmer/lib/environment.js]: this doesn't work for `inverse`

```hbs
{{#my-component}}  default {{else}} inverse {{/my-component}}
```

File: packages/ember-glimmer/lib/environment.js [hunk 3]
```diff
@@ -224,12 +227,15 @@ export default class Environment extends GlimmerEnvironment {
     return false;
   }
 
-  getComponentDefinition(path) {
+  getComponentDefinition(path, parentMeta = {}) {
     let name = path[0];
     let definition = this._components[name];
```
Comment (krisselden) [inline packages/ember-glimmer/lib/environment.js]: The name needs to be the resolved name (not the local name) before looking up the cached compiled definition. We need to refactor to address this, maybe use the _debugContainerKey or less debug friendly, use guidFor(ComponentClass) as the definition key.

**Comment by asakusuma** [comment]
@krisselden blueprints is failing for some reason. everything else passes

`npm ERR! code 1`

**Comment by rwjblue** [comment]
This looks good to me.

@asakusuma - Thanks for working on this, it has been a looooonnnnggggg road... 😸 

@krisselden - r?

**Pull request merged**